core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -343,6 +359,50 @@ private[spark] class IndexShuffleBlockResolver(
     }
   }
 
+  /**
+   * This is only used for reading local merged block data. In such cases, all chunks in the
+   * merged shuffle file need to be identified at once, so the ShuffleBlockFetcherIterator
+   * knows how to consume local merged shuffle file as multiple chunks.
+   */
+  override def getMergedBlockData(blockId: ShuffleBlockId): Seq[ManagedBuffer] = {
+    val indexFile = getMergedBlockIndexFile(conf.getAppId, blockId.shuffleId, blockId.reduceId)
+    val dataFile = getMergedBlockDataFile(conf.getAppId, blockId.shuffleId, blockId.reduceId)
+    // Load all the indexes in order to identify all chunks in the specified merged shuffle file.
+    val size = indexFile.length.toInt
+    val buffer = ByteBuffer.allocate(size)
+    val offsets = buffer.asLongBuffer
+    val dis = new DataInputStream(Files.newInputStream(indexFile.toPath))
+    try {
+      dis.readFully(buffer.array)
+    } finally {
+      dis.close()
+    }
```

Comment:
Use `tryWithResource` instead.
Something like
```
val offsets = Utils.tryWithResource {
  new DataInputStream(Files.newInputStream(indexFile.toPath))
} { dis =>
  val buffer = ByteBuffer.allocate(size)
  dis.readFully(buffer.array)
  buffer.asLongBuffer
}
```

Reply:
This is not resolved? @zhouyejoe

Reply:
On second thought, given `Utils` is not available, there might not be an alternative location for this.
We might need to look at a common infra module for utilities like this which all modules can depend on ... thoughts @Ngone51 ?

Reply:
Updated as @mridulm suggested.